node/task.ts

```diff
@@ -1116,6 +1204,20 @@ export function _loadData(): void {
         }
     }
     debug('loaded ' + loaded);
+
+    // store public variable metadata
+    JSON.parse(process.env['VSTS_PUBLIC_VARIABLES'] || '[]')
```

Comment:
I typically protect JSON.parse because it can throw but I guess this isn't user data (config file etc...).  We should always write valid JSON if it's set ...